Content.Server/Changeling/ChangelingSystem.cs

```diff
@@ -0,0 +1,1114 @@
+    {
+        var newComp = EnsureComp<ChangelingComponent>(target);
+        newComp.AbsorbedDNA = comp.AbsorbedDNA;
+        newComp.AbsorbedDNAIndex = comp.AbsorbedDNAIndex;
+        newComp.Chemicals = comp.Chemicals;
+
+        newComp.IsInLesserForm = comp.IsInLesserForm;
+        newComp.CurrentForm = comp.CurrentForm;
+
+        newComp.TotalAbsorbedEntities = comp.TotalAbsorbedEntities;
+        newComp.TotalStolenDNA = comp.TotalStolenDNA;
+
+        return comp;
+    }
+    private EntityUid? TransformEntity(EntityUid uid, TransformData? data = null, EntProtoId? protoId = null, ChangelingComponent? comp = null, bool persistentDna = false)
+    {
+        EntProtoId? pid = null;
+
+        if (data != null)
+        {
+            if (!_proto.TryIndex(data.Appearance.Species, out var species))
+                return null;
+            pid = species.Prototype;
+        }
+        else if (protoId != null)
+            pid = protoId;
+        else return null;
+
+        var config = new PolymorphConfiguration()
+        {
+            Entity = (EntProtoId) pid,
+            TransferDamage = true,
+            Forced = true,
+            Inventory = PolymorphInventoryChange.Transfer,
+            RevertOnCrit = false,
+            RevertOnDeath = false
+        };
+        var newUid = _polymorph.PolymorphEntity(uid, config);
+
+        if (newUid == null)
+            return null;
+
+        var newEnt = newUid.Value;
+
+        if (data != null)
+        {
+            Comp<FingerprintComponent>(newEnt).Fingerprint = data.Fingerprint;
+            Comp<DnaComponent>(newEnt).DNA = data.DNA;
+            _humanoid.CloneAppearance(data.Appearance.Owner, newEnt);
+            _metaData.SetEntityName(newEnt, data.Name);
+            var message = Loc.GetString("changeling-transform-finish", ("target", data.Name));
+            _popup.PopupEntity(message, newEnt, newEnt);
+        }
+
+        RemCompDeferred<PolymorphedEntityComponent>(newEnt);
+
+        if (comp != null)
+        {
+            // copy our stuff
+            var newLingComp = CopyChangelingComponent(newEnt, comp);
+            if (!persistentDna && data != null)
+                newLingComp?.AbsorbedDNA.Remove(data);
+            RemCompDeferred<ChangelingComponent>(uid);
+
+            if (TryComp<StoreComponent>(uid, out var storeComp))
+            {
+                var storeCompCopy = _serialization.CreateCopy(storeComp, notNullableOverride: true);
+                RemComp<StoreComponent>(newUid.Value);
+                EntityManager.AddComponent(newUid.Value, storeCompCopy);
+            }
+        }
+
+        // exceptional comps check
+        // there's no foreach for types i believe so i gotta thug it out yandev style.
+        if (HasComp<HeadRevolutionaryComponent>(uid))
+            EnsureComp<HeadRevolutionaryComponent>(newEnt);
+        if (HasComp<RevolutionaryComponent>(uid))
+            EnsureComp<RevolutionaryComponent>(newEnt);
```

Comment:
Do the same thing with MindShield comp. Lings can't copy it, but also can't just destroy